crowbar_framework/app/models/service_object.rb

```diff
@@ -919,6 +919,36 @@ def self.proposal_to_role(proposal, bc_name)
     RoleObject.new role
   end
 
+  #  we can speed-up the application of (n+1)th role if both(n,n+1)
+  #  roles are being applied on the same node
+  #
+  #  eg. In our 2node deployment ceilometer{server,central} are always
+  #  applied on the same node, given that they have different priorities,
+  #  they are to be applied, one after the other, these priorities come
+  #  from element_run_list_order
```

Comment:
Two nitpicks here:

1. The priorities come from `chef_order` by default but can be overridden per role by `element_run_list_order`
2. `ceilometer-{server,central}` is just one example, but this code will merge any adjacent batches regardless of what run_list priority the roles in.  Is it possible to make the code automatically check that the run_list priority of batch `n` is lower than the priority of batch `n+1`, and only merge if that is true?  If that was added then I think this would feel safe enough to me for a +1 vote.